Design Refresh & React/TypeScript Templates

paiziak-o opened the pull request:
Demo: https://fullstackmb-github-io-3t2e.vercel.app/

Before merging, could you review all the text? I've rephrased it.
Regarding the text in the footer, I would really like to keep the email for "Contact Us." However, if that's not possible, maybe we can include a list of leadership with links to LinkedIn.

TODO after:

- Fix Playwright tests.
- Clean up eleventy.config.ts, as the default hot reloader didn't work very well with subcomponents.
- Small UI fixes
- Split css on small files (component based)
- Move style attribute from JSX code to CSS (part of the code was generated by AI)
- Unit tests
- Code cleaning

davidwesst commented:
This looks really good. Only thing I found is that the e2e tests are no longer working. Quick glance at the report seems like adjustments to the tests need to be handled rather than the code itself.

Beyond that-- the version can be bumped to 3.1 to capture these changes.

paiziak-o commented:
any thoughts about contact us in the footer?

paiziak-o commented:
@davidwesst fixed e2e tests and bumped version

paiziak-o commented:
Changed the labels for two menu items:
- "Join" → "Join Slack"
- "RSVP" → "Meetup"

![image](https://github.com/user-attachments/assets/d4e5ee12-ffc6-44fa-9a22-4e06558cc0fc)

Additionally, added extra links to the "Contact Us" section:
![image](https://github.com/user-attachments/assets/8b3b413b-dac9-46e5-b2cf-17b43dbd29f1)

Please let me know your thoughts.

davidwesst commented:
This looks solid. Going to mark it as approved-- but I did notice one thing that isn't a show stopper but want to get your thoughts before I merge the PR.

In a desktop resolution, the "upcoming event" images looks like they are being squished a bit (i.e. their width is being shortened, and changing the original aspect ratio of the image) which looks odd compared to the main image. Probably only really noticeable when we have an upcoming code and coffee and then the image is visible again in the upcoming events section.

Let me know your thoughts on if we should fix it with this PR or merge this and create a new bug for it.

paiziak-o commented:
Yes, I see there is a fixed width and height, which is not ideal.

However, I personally don't see an easy CSS solution for the squished issue. We need to maintain the height at 200px to keep the design visually nice, and I think the width should be set to "auto" to allow it to resize images properly. For such small blocks, it could be tricky to implement.

If possible, I would prefer to merge it as is and report it on GitHub as a bug, allowing others to contribute if they wish.

@davidwesst what do you think?
